lib/ssh/internal_doc/ssh_notes.md

```diff
@@ -1,36 +1,108 @@
+# SSH supervision tree (prototype)
 ```mermaid
 ---
-title: SSH supervision tree
+title: SSH supervision tree (prototype)
 ---
 flowchart RL
-    d_sup --> sup[["ssh_sup\n[o4o]"]]
+    d_sup --> sup[["ssh_sup\n(ssh_app.erl)\n[o4o]"]]
     c_sup --> sup
 
     subgraph client
-    system_sup --> c_sup[["sshc_sup\n[o4o]"]]
-
+    subsystem_sup --> c_sup[["sshc_sup\n(ssh1_app.erl)\n[o4o]\nauto_shutdown=never"]]
     subgraph connection_c
-    subsystem_sup --> system_sup[["ssh_system_sup\n[o4o]"]]
-    connection_handler["ssh_connection_handler"] --> subsystem_sup[["ssh_subsystem_sup\n[o4a]"]]
-    channel_sup[["ssh_channel_sup\no4o"]] --> subsystem_sup
+    connection_handler["ssh_connection_handler\nSIGNIFICANT"] --> subsystem_sup[["ssh_subsystem_sup\n[o4a]\nauto_shutdown=any_significant"]]
+    channel_sup[["ssh_channel_sup\n[o4o]"]] --> subsystem_sup
+    sftp["ssh_sftp"] --> channel_sup
     tcpip_forward_acceptor_sup[["ssh_tcpip_forward_acceptor_sup\n[o4o]"]] --> subsystem_sup
+    ssh_tcpip_forward_acceptor["ssh_tcpip_forward_acceptor"] --> tcpip_forward_acceptor_sup
+    end
     end
 
+    subgraph server
+    system_sup_s --> d_sup[["sshd_sup\n(ssh_app.erl)\n[o4o]"]]
+    acceptor_sup --> system_sup_s[["ssh_system_sup\n[o4o]\nauto_shutdown=all_significant"]]
+    acceptor["ssh_acceptor"] --> acceptor_sup[["ssh_acceptor_sup\n[o4o]\nSIGNIFICANT"]]
+
+    subsystem_sup_s --> system_sup_s
+
+    subgraph connection_s
+    connection_handler_s["ssh_connection_handler\nSIGNIFICANT"] --> subsystem_sup_s[["ssh_subsystem_sup\n[o4a]\nauto_shutdown=any_significant\nSIGNIFICANT"]]
+    channel_sup_s[["ssh_channel_sup\n[o4o]"]] --> subsystem_sup_s
+    tcpip_forward_acceptor_sup_s[["ssh_tcpip_forward_acceptor_sup\n[o4o]"]] --> subsystem_sup_s
+    ssh_tcpip_forward_acceptor_s["ssh_tcpip_forward_acceptor"] --> tcpip_forward_acceptor_sup_s
+    sftd1["ssh_sftpd"] --> channel_sup_s
+    end
+    end
+```
+## Questions
+1. ssh_subsystem_sup - one_for_all and auto_shutdown ... any process crash results with ssh_subsystem_sup shutdown?
```

Comment:
Short answer: No :wink:

Long answer: see [EEP-56](https://www.erlang.org/eeps/eep-0056#rationale):
> Otherwise, the following rules apply when a significant child exits _on its own_:
>
> * A `transient` child will be restarted (not cause a supervisor shutdown) if it exits abnormally. If it exits normally…
       * if the supervisor `auto_shutdown` flag is `any_significant`, the supervisor will shut down
       * if the supervisor `auto_shutdown` flag is `all_significant`, the supervisor will shut down if the child was the last active significant child
>
> * A `temporary` child will never be restarted. If it exits normally or abnormally, the same rules as for `transient` children apply, in regard to the supervisor `auto_shutdown` flag.
>
> [...]
>
> To be clear, the above rules only apply when _significant_ children exit by _themselves_, that is, not when being terminated manually via `supervisor:terminate_child/2`, not when other non-significant children exit, and not when being terminated as a consequence of a sibling’s death in the `one_for_all` or `rest_for_one` strategies.

Reply:
yes. this is just a scratchpad ...
thanks for commenting anyway :-)